src/neuroglancer/layer_panel.ts

```diff
@@ -354,31 +355,33 @@ export class LayerPanel extends RefCounted {
   private scheduleUpdate = this.registerCancellable(animationFrameDebounce(() => this.update()));
 
   private update() {
-    this.valueUpdateNeeded = false;
-    this.updateLayers();
-    let values = this.manager.layerSelectedValues;
-    for (let [layer, widget] of this.layerWidgets) {
-      let userLayer = layer.layer;
-      let text = '';
-      if (userLayer !== null) {
-        let value = values.get(userLayer);
-        if (value !== undefined) {
-          value = Array().concat(value);
-          value = value.map((x: any) => {
-            if (x === null) {
-              return 'null';
-            } else if (Math.fround(x) === x) {
-              // FIXME: Verify actual layer data type
-              return float32ToString(x);
-            } else {
-              return x;
-            }
-          });
-          text += value.join(', ');
+    if (this.displayLayerItemValue.value === true){
```

Comment:
You could simplify this nesting by just returning early if the value is false.  Also I think we still need to call updateLayers() in case the layer names changed.

Reply:
Also, if you update an attribute of the layer panel element, e.g. this.element.dataset.showHoverValues = this.displayLayerItemValue.value.toString() , then you can use a CSS rule to hide that area of the layer widget.

Reply:
> Also, if you update an attribute of the layer panel element, e.g. this.element.dataset.showHoverValues = this.displayLayerItemValue.value.toString() , then you can use a CSS rule to hide that area of the layer widget.

I am not sure whether I properly understand what you are suggesting. Would you like to handle the entire hiding of the value display by switching between css classes? In this case the whole loop in update() would still be executed when hovering across the segementation which seems inefficient or at least unnecessary. Thanks!

Reply:
I was suggesting to keep your current logic of not updating the values if they are hidden, but additionally use the CSS rules so that no space is taken up --- exactly as you have done.